File: YuGiOh2/Cards/C22046459.cs
using System.Linq;
using YuGiOh2.Base;

namespace YuGiOh2.Cards
{
    /// <summary>
    /// 巨大化
    /// </summary>
    public class C22046459
    {
        public static int Type { get; } = (int)ChooseTargetType.AllFaceUpMonster;

        public static bool CheckIfAvailable(Player player)
        {
            return player.Field.MonsterFields.Any(c => c != null && !c.Status.FaceDown) ||
                player.Enemy.Field.MonsterFields.Any(c => c != null && !c.Status.FaceDown);
        }

        public static void ProcessEffect(string targetID, Player player)
        {
            if (targetID == null)
                return;

            MonsterCard target = player.Field.MonsterFields.FirstOrDefault(c => c != null && c.UID == targetID) ??
                player.Enemy.Field.MonsterFields.FirstOrDefault(c => c != null && c.UID == targetID);
            if (target == null || player.HP == player.Enemy.HP)
                return;
            target.ATK = player.HP > player.Enemy.HP ? target.ATK >> 1 : target.ATK << 1;
        }
    }
}

File: YuGiOh2/Scripts/C22046459.lua
Type = 13;
--巨大化
function checkIfAvailable(player)
    for i, monster in pairs(player.Field.MonsterFields) do
        if monster ~= nil and not monster.Status.FaceDown then
            return true;
        end
    end
    for i, monster in pairs(player.Enemy.Field.MonsterFields) do
        if monster ~= nil and not monster.Status.FaceDown then
            return true;
        end
    end
    return false;
end

function processEffect(targetID, player)
    if targetID == nil then
        return ;
    end
    local target = nil;
    for i, monster in pairs(player.Field.MonsterFields) do
        if monster ~= nil and not monster.Status.FaceDown and monster.UID == targetID then
            target = monster;
        end
    end
    for i, monster in pairs(player.Enemy.Field.MonsterFields) do
        if monster ~= nil and not monster.Status.FaceDown and monster.UID == targetID then
            target = monster;
        end
    end
    if target == nil or player.HP == player.Enemy.HP then
        return ;
    end
    if player.HP > player.Enemy.HP then
        target.ATK = target.ATK / 2;
    else
        target.ATK = target.ATK * 2;
    end
end